### PR TITLE
Bugfix/update tutorial scenarios

### PR DESCRIPTION
**Pull request by MarleneBusch**
# Description

- Fixed the tutorials by updating their scenario
- Fixed a link for Read The Docs
- Fixed a typo in one Read The Docs tutorial

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by MarleneBusch** [comment]
This pull request now also contains:
- scenario generators in the tutorials, without ARTIST_ROOT
- updated data files in tests/ and tutorials/
- a new setup for distributed computation for ddp that should work on all systems
- major bugfixes in the heliostat-tracer, that fix issue #97

**Review by kalebphipps — APPROVED** [review]
Thanks for the PR - very clean code and I like the changes to the tutorials!

I have locally removed ``psutil`` from the ``pyproject.toml`` since I don't think we use it anymore -- if I have missed something and the tests suddenly fail I will add it back again.

I have also locally made a few small changes to the docs, correcting three or four typos and one broken link. I will push them now.

I am happy for this to be merged now - but if you want to add in the logging message I suggested for when certain Ranks are left idle that would be great. I am happy for you to directly merge in to ``ARTIST`` main once this logging is there (or if you disagree and think the logging is a waste of time). Therefore, this PR is officially approved and I don't require an additional review :)

File: artist/raytracing/heliostat_tracing.py
[the remark this answers is not in the document]
Comment (MarleneBusch) [reply artist/raytracing/heliostat_tracing.py:1]: The changes made concerned three seperate issues of the old raytracer. 
- First, something we didnt even look for but noticed by chance. The part about the soft binning  behaved unexpected. The theroy behind the soft binning aims to continously distribute rays to the 4 surrounding pixels. You would expect that if a ray that hits the target inbetween 4 pixels but is closest to the upper right pixel, that this upper right pixel would receive the highest intensity from this ray. This did not happen, as the assignment from ray intersection to pixel was in an incorrect order. I doubt that we could have perceived this error just by looking at output bitmaps, especially with the 256x256 resolution we most commonly use. The difference of the incorrect order is very small and maybe even irrelevant to flux predictions of whole fields but since it simply does not align with physics or realistic modeling, fixing it was important to me.
- Next, the initialisation of the empty final_bitmap was wrong, and caused the bitmap to be transposed. When considering 2d images in an (E,U) coordinate-system, the resolution in E is determined by the number of values in the U dimension. This had to be corrected in the initialization of the bitmap:. 
`        total_bitmap = torch.zeros(
            [self.bitmap_resolution_u, self.bitmap_resolution_e],
            dtype=dx_ints.dtype,
            device=device,
        )
`
- The issues of the bitmaps being mirrored bith horizontally and vertically was caused by the way the bitmap was filled. The horizontal flip happened because tensor indices start with (0,0) in the upper left corner and their indices increase going further down, while our coordinates have their origin a lower left corner and axis increases going upwards. We fixed this by:
`self.bitmap_resolution_u - 1 - y_inds[indices]` in the `total_bitmap.index_put_()` method. The vertical flip happened because contradicting to my (and i guess most peoples) intuition the bitmaps created by the raytracer have to be looked at from behind the receiver. Upon reflecting this issue it makes sense, as we have defined a right handed (E,N,U) coordinate system in which the raytracer creates bitmaps. When standing behind the tower and looking straight ahead (to the north) then east is to your right. When you change your position to somewhere in the field, because you want to see the flux on the target, and you look directly into the south, east will be to your left, and you have to flip that axis. In the code this was corrected by: `self.bitmap_resolution_e - 1 - x_inds[indices],`

File: artist/util/alignment_optimizer.py [hunk 1]
```diff
@@ -333,21 +333,22 @@ def _optimize_kinematic_parameters_with_raytracing(
...
+            if self.is_distributed:
+                torch.distributed.all_reduce(
                     final_bitmap, op=torch.distributed.ReduceOp.SUM
                 )
 
             final_bitmap = raytracer.normalize_bitmap(final_bitmap)
 
             center = utils.get_center_of_mass(
-                bitmap=torch.flip(final_bitmap, dims=(0, 1)),
+                bitmap=final_bitmap,
```
[the remark this answers is not in the document]
Comment (MarleneBusch) [reply artist/util/alignment_optimizer.py:351]: :) yes!

File: artist/raytracing/heliostat_tracing.py [hunk 2]
```diff
@@ -91,11 +91,117 @@ def __getitem__(self, idx: int) -> tuple[torch.Tensor, torch.Tensor]:
...
+        """
+        super().__init__(dataset)
+        self.number_of_samples = len(dataset)
+        self.world_size = world_size
+        self.rank = rank
+        self.shuffle = shuffle
+        self.seed = 0
+
+        # Adjust num_replicas if dataset is smaller than world_size
+        self.active_replicas = min(self.number_of_samples, self.world_size)
```
[the remark this answers is not in the document]
Comment (MarleneBusch) [reply artist/raytracing/heliostat_tracing.py:168]: done
